crates/driver/src/infra/solver/mod.rs

```diff
@@ -107,7 +111,13 @@ impl Solver {
             "application/json".parse().unwrap(),
         );
         headers.insert(reqwest::header::ACCEPT, "application/json".parse().unwrap());
-        // TODO(#907) Also add an auth header
+
+        for (key, val) in config.request_headers.iter() {
+            if let Ok(header_name) = HeaderName::try_from(key) {
```

Comment:
Since this is part of the initialization logic instead of silently ignoring errors and possibly running with an unexpected behavior we should panic with a helpful error message if the header name or value could not be parsed.